### ISSUE TITLE
[java] UseIOStreamsWithApacheCommonsFileItem: false negative with `this` pointer 

### ISSUE THREAD

**Issue by mohui1999** [pmd/pmd#4421]
**Affects PMD Version:** 6.55.0

**Rule:** [UseIOStreamsWithApacheCommonsFileItem](https://pmd.github.io/latest/pmd_rules_java_performance.html#useiostreamswithapachecommonsfileitem)

**Description:**

Using `globalFileItem.get()` is bad here. So PMD would report **UseIOStreamsWithApacheCommonsFileItem** here. But when using `this.globalFileItem.get()`, it does not report.

**Code Sample demonstrating the issue:**
Maybe a false negative:

```java
import org.apache.commons.fileupload.FileItem;
public class Foo {
    private FileItem globalFileItem;
    private byte[] get(final FileItem fileItem) {
        FileItem sameFileItem = fileItem;     
        return sameFileItem.get();      // report here
    }
    private byte[] get() {
        this.globalFileItem.get();    // NOT report here
    }
}
```

Correctly reported code:

```java
import org.apache.commons.fileupload.FileItem;
public class Foo {
    private FileItem globalFileItem;
    private byte[] get(final FileItem fileItem) {
        FileItem sameFileItem = fileItem;
        return sameFileItem.get(); // report here
    }
    private byte[] get() {
        globalFileItem.get(); // report here
    }
}
```

**Expected outcome:**

PMD should report a violation at line 12, but doesn't. This is a false negative.

**Running PMD through:** *[CLI]*

**Comment by mohui1999** [comment]
Fixed in PMD 7.0
